sklearn/datasets/_base.py

```diff
@@ -1424,7 +1427,7 @@ def _sha256(path):
     return sha256hash.hexdigest()
 
 
-def _fetch_remote(remote, dirname=None):
+def _fetch_remote(remote, dirname=None, n_retries=3, delay=1):
```

Comment:
Could you please add a unit tests that calls `_fetch_remote` on an instance of `RemoteFileMetadata` with an invalid URL (`"https://scikit-learn.org/this_file_does_not_exist.pkl"`) and checks that 3 warnings are raised and then finally an `HTTPError` with code 404 is raised.

Reply:
Hi @ogrisel , I've added the test cases as you suggested :)